pages/docs/database-connections/role-based-credentials.mdx

```diff
@@ -0,0 +1,70 @@
+---
+description: Role-based data credentials allow you to control what users can access from the same data connection.
+---
+
+import { Callout } from "nextra-theme-docs";
+
+# Role-Based Database Credentials
+
+All Hashboard data connections can be optionally configured to use additional credentials which can be assigned to users on a per-role basis. 
+
+After setting up database-level permissions within the data warehouse you use with Hashboard, these credentials can be added within Hashboard and then assigned to your Hashboard project's roles.
+
+This enables you to control what data Hashboard users can access when viewing models, saved explorations, and dashboards built from the same data connection.
+Role-based credentials can be used to restrict access to certain tables, or even certain rows or columns within a table, based on a user's role.
+
+### Adding credentials
+
+The first step to adding role-based credentials is to create a [database connection](/docs/database-connections).
+
+Note that the credentials specified in the top-level of the database connection form is the *default* credential. 
+That is, unless specified otherwise, all users will connect via the information provided in the database connection form.
+
+<Callout>
+    If a user doesn't have any specified credentials within their roles, they will use the default credential for the data connection.
+    Therefore, it's best practice for the default credential to be the most restrictive credential permissions-wise.
+</Callout>
+
+At the bottom of the data connection form, you'll see a section for **Role-based credentials**.
+
+Click **Add credential** and fill out the fields provided. Give the credential a unique and descriptive name, 
+as it will be used to assign to roles later on.
+
+Once you're done, make sure to **Test** the credentials before saving to ensure they are connecting properly.
+
+Finally, click **Save**.
+
+#### Settings
```

Comment:
I would rename this to `Credential fields` and reference it in the `Click **Add credential** and fill out...` step above (e.g. see `Credential fields` below for more details on ...`).

Alternatively, you could just include it under that bullet.